oauth2_http/javatests/com/google/auth/oauth2/UserAuthorizerTest.java

```diff
@@ -357,6 +396,34 @@ public void getCredentialsFromCode_conevertsCodeToTokens() throws IOException {
     assertEquals(GRANTED_SCOPES, credentials.getAccessToken().getScopes());
   }
 
+  @Test
+  public void getCredentialsFromCode_additionalParameters() throws IOException {
+    MockTokenServerTransportFactory transportFactory = new MockTokenServerTransportFactory();
+    transportFactory.transport.addClient(CLIENT_ID_VALUE, CLIENT_SECRET);
+
+    Map<String, String> additionalParameters = new HashMap<String, String>();
+    additionalParameters.put("param1", "value1");
+    additionalParameters.put("param2", "value2");
+
+    transportFactory.transport.addAuthorizationCode(
+        CODE, REFRESH_TOKEN, ACCESS_TOKEN_VALUE, GRANTED_SCOPES_STRING, additionalParameters);
+    TokenStore tokenStore = new MemoryTokensStorage();
+    UserAuthorizer authorizer =
+        UserAuthorizer.newBuilder()
+            .setClientId(CLIENT_ID)
+            .setScopes(DUMMY_SCOPES)
+            .setTokenStore(tokenStore)
+            .setHttpTransportFactory(transportFactory)
+            .build();
+
+    UserCredentials credentials =
+        authorizer.getCredentialsFromCode(CODE, BASE_URI, additionalParameters);
```

Comment:
maybe add at least an after test without params? to check they aren't saved in any property